app/services/state_file/send_pre_deadline_reminder_service.rb

```diff
@@ -0,0 +1,46 @@
+module StateFile
+  class SendPreDeadlineReminderService
+    BATCH_SIZE = 10
+    HOURS_AGO = 24
+
+    def run
+      cutoff_time_ago = HOURS_AGO.hours.ago
+      intakes_to_notify = []
+      ApplicationRecord::STATE_INTAKE_CLASS_NAMES.map do |base_class|
```

Comment:
Probably could use `each` rather than `map` here. Unless you want to do...
```
intakes_to_notify = ApplicationRecord::STATE_INTAKE_CLASS_NAMES.map do |base_class|
  base_class.constantize.left_joins( ...
end.flatten
```

Reply:
Good catch, done.